Modules/IO/NIFTI/test/itkNiftiImageIOTest.h

```diff
@@ -57,7 +57,7 @@ constexpr unsigned char SUPERIOR = 32; /*Bit pattern 0 0 1  00000*/
 // The only specializations of this function template that may be called are given below.
 template <unsigned int TDimension>
 typename itk::ImageBase<TDimension>::DirectionType
-PreFillDirection() = delete;
+PreFillDirection();
 
 template <>
```

Comment:
With the `= delete` removed above, it becomes an ordinary (template) declaration.  I am thinking that the four declarations of the template specializations that follow it do not do anything now and can be removed.  In particular, removal does not cause any problems when I modify your [example](https://godbolt.org/z/W58nEWadM) accordingly.

Reply:
@Leengit It may be fine to remove the four declarations of the template specializations from my [example](https://godbolt.org/z/W58nEWadM) but I'm not sure if we can just remove them from the "itkNiftiImageIOTest.h", while having them defined in the corresponding cxx file, "itkNiftiImageIOTest.cxx", and being called from yet two other cxx files (itkNiftiImageIOTest3.cxx and itkNiftiImageIOTest7.cxx), which only "see" the header file, during compilation 🤷